src/main/java/seedu/address/petui/PetCard.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.petui;
+
+import java.util.Comparator;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.pet.Pet;
+import seedu.address.ui.UiPart;
+
+/**
+ * An UI component that displays information of a {@code Pet}.
+ */
+public class PetCard extends UiPart<Region> {
+
+    private static final String FXML = "PetListCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    public final Pet pet;
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label name;
+    @FXML
+    private Label id;
+    @FXML
+    private Label gender;
+    @FXML
+    private Label dateOfBirth;
+    @FXML
+    private Label species;
+    @FXML
+    private FlowPane tags;
+
+    public PetCard(Pet pet, int displayedIndex) {
+        super(FXML);
+        this.pet = pet;
+        id.setText(displayedIndex + ". ");
+        name.setText(pet.getName().toString());
+        gender.setText(pet.getGender().toString());
+        dateOfBirth.setText(pet.getDateOfBirth().toString());
+        species.setText(pet.getSpecies().toString());
+        pet.getTags().stream()
+                .sorted(Comparator.comparing(tag -> tag.tagName))
+                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof PetCard)) {
+            return false;
+        }
+
+        // state check
+        PetCard card = (PetCard) other;
+        return id.getText().equals(card.id.getText())
```

Comment:
Since we only using pet names to identify the pets, should we just let pet class do the comparison here also?

Reply:
The line below has pet class comparison. I think we leave PetCard id comparison here to ensure correctness at Ui level since id is attribute of PetCard